src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -551,10 +551,15 @@ struct
     in
     let root_ledger = Staged_ledger.ledger root in
     let soon_to_be_root_ledger = Staged_ledger.ledger soon_to_be_root in
-    let soon_to_be_root_merkle_root =
-      Ledger.merkle_root soon_to_be_root_ledger
+    let%bind soon_to_be_root_merkle_root =
+      Deferred.create (fun ivar ->
+          Ivar.fill ivar @@ Ledger.merkle_root soon_to_be_root_ledger )
```

Comment:
this eagerly computes a filled ivar, ie, is a fancy no-op.

Reply:
Does it? I thought all the arguments would be evaluated before applying the fill function.